src/smackw32/smackw32.c

```diff
@@ -9,55 +9,236 @@
 #include "harness/hooks.h"
 // lib/libsmacker
 #include "smacker.h"
+// lib/miniaudio
+#include <miniaudio/miniaudio.h>
 
-uint32_t smack_last_frame_time;
+extern ma_engine miniaudio_engine;
```

Comment:
This re-uses `miniaudio_engine`, so the separation between smackw32 and S3 is not perfect.